seoul_09_kyoohyunkim/BOJ/Gold III/boj 10986/Boj_10986.java

```diff
@@ -0,0 +1,38 @@
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.StringTokenizer;
+
+
+public class Boj_10986 {
+
+    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+    static StringTokenizer st;
+    public static void main(String[] args) throws Exception {
+        st = new StringTokenizer(br.readLine());
+        int n = Integer.parseInt(st.nextToken());
+        int m = Integer.parseInt(st.nextToken());
+
+        long[] arr = new long[n + 1];
+        st = new StringTokenizer(br.readLine());
+
+        int[] mods = new int[m];
+
+        for (int i = 1; i <= n; i++) {
```

Comment:
for문 부분을 이렇게도 하는 것 같더라구요
```suggestion
while (n-->0) {
            int cur = Integer.parseInt(st.nextToken());
            sum += cur;
            sum %= m;
            answer += cnt[sum];
            cnt[sum]++;
            if (sum==0) answer++;
        }
```
가독성이 좋길래 참고용으로 보시면 좋을 것 같아 가져와봤습니다 😀

Reply:
저도 수학 문제는 너무 어렵네요 ㅠㅠ 운이 좋아서 풀 수 있었던 것 같습니다.
알고리즘에서는 크게 가독성을 신경 쓰지는 않는데, 여기 그룹에 올릴 때에는 더 신경 써서 작성하는 편이 좋겠네요